### DISCUSSION TITLE
Vitest integration tests and $env/dynamic/private

### DISCUSSION THREAD

**Discussion by kyllerss** [sveltejs/kit#8169] [Q&A]
I finally got around to migrating away from `import.meta.env.*` for all of my environment variables and onto the SvelteKit-recommended `$end/dynamic|static/public|private`. I had a number of integration tests I had written in `vitest` that are now breaking with the following error:

```
FAIL  tests/integration/hooks.test.ts [ tests/integration/hooks.test.ts ]
Error: Cannot import $env/dynamic/private into client-side code
 ❯ Context.load node_modules/@sveltejs/kit/src/exports/vite/index.js:304:12
 ❯ Object.load node_modules/vite/dist/node/chunks/dep-67e7f8ab.js:40201:46
 ❯ loadAndTransform node_modules/vite/dist/node/chunks/dep-67e7f8ab.js:36553:24
```
My integration tests import a number of back-end modules (eg. database access code, etc...).

I am surprised by the fact that the integration tests are being treated as client code that is tripping up the server-client serialization safety mechanism.

I am at a loss as to what to do here. Any recommendations on how to address this issue? Is there any way I can turn off this safety mechanism for my integration tests?

**Comment by kyllerss** [comment]
Created a bug report [here](https://github.com/sveltejs/kit/issues/8180).

**Reply by DoisKoh** [reply]
Just chiming in to say I have the same issue as well.

**Comment by MahmoodKhalil57** [comment]
I also have the same issue
